.github/workflows/release-spec-prod.yml

```diff
@@ -0,0 +1,23 @@
+name: '(PROD) Release OpenAPI Spec'
+on:
+  workflow_dispatch: {}
+  schedule:
+    - cron: '0 9 * * *' # Run at 9AM UTC
+permissions:
+  contents: write
+jobs:
+  run-validations:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
+      - name: Setup Node.js
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b
+        with:
+          node-version: 20
+      - name: Install Spectral
+        run: npm install -g @stoplight/spectral-cli@6.11.1 && spectral --version
+      - name: Run Spectral
+        run: spectral lint ./tools/cli/test/data/apiregistry_spec.{json,yml,yaml} --ruleset=.spectral.yaml # we will update this to lint the FOAS in CLOUDP-263186
```

Comment:
[nit] can we generate a foas and store there? just so that we can already start checking the full spec! also ok to do in a follow-up

Reply:
foas using `tools/cli/test/data/openapi-mms.json` - also, can we make sure to capture x-xgen-sha in that file? (I can't open, but I guess its there!)

Reply:
doing this in the follow-up PR 👍

Reply:
For maintainability my suggestion would be to always check federated spec as well.
Spectral will be our early canary in the mine for any potential federation issues.

Reply:
https://github.com/mongodb/openapi/pull/97